### PR TITLE
Update Finnish translations

### PR DESCRIPTION
**Pull request by valtterikantanen**
I've made several updates to the Finnish translations, aimed at improving accuracy and consistency. Key changes include:

### Typographical corrections
- Corrected typos in specific terms. Examples include:
  - From *kanditaatintutkielma* to **kandidaatintutkielma** (bachelor's thesis and candidate thesis)
  - From *Euroopan Unioni* to **Euroopan unioni** (European Union), adhering to Finnish capitalization rules.

### Terminology updates
- Updated the term for European patent to **eurooppapatentti**, reflecting the accurate Finnish terminology as these patents are granted by the European Patent Office, not the European Union. This aligns with the terminology database at [PRH](http://uri.suomi.fi/terminology/prh/c80).
- Revised the abbreviation for *patenttihakemus* (patent application) from *pat. hak.* to **pat.hak.**, maintaining consistency with the absence of a space in the original term.
- Modified the term for doctoral thesis from *tohtorinväitöskirja* to **väitöskirja**. *Väitöskirja* is a lot more common in this context and it can only mean a doctoral thesis, as other thesis types are called *tutkielma* in Finnish.

### Translation of "et al."
- Translated the Latin term *et al.* into the Finnish **ym.**, to better reflect Finnish language practices. This change, while potentially controversial, is supported by several Finnish academic sources and guidelines, indicating a preference for Finnish terms over Latin when possible. These sources include:
  - The scientific journal [*Psykologia*](http://www.psykologia.fi/kirjoittajille/yleiset-ohjeet-kirjoittajille), published by the Finnish Psychological Society.
  - [*Yliopistopedagogiikka*](https://yliopistopedagogiikka.files.wordpress.com/2022/10/yo-peda-tieteelliset-kirjoittajaohjeet-2019-1.pdf), a journal on university pedagogy.
  - A [guide by the University of Helsinki](https://blogs.helsinki.fi/kielijelppi/lahdeviitteiden-merkinta/) on citation practices.
  - The Institute for the Languages of Finland's [guide](https://kielitoimistonohjepankki.fi/ohje/lyhenneluettelo-e/) which lists *ym.* as the Finnish equivalent of *et al.*

It's worth noting that similar translations for "et al." have been adopted in the Swedish and German versions of BibLaTeX, as evidenced by their respective `.lbx` files:

https://github.com/plk/biblatex/blob/2a9dddb2f371305fcbdc9392d92457571e91a9d0/tex/latex/biblatex/lbx/german.lbx#L350
https://github.com/plk/biblatex/blob/2a9dddb2f371305fcbdc9392d92457571e91a9d0/tex/latex/biblatex/lbx/swedish.lbx#L309-L310

### CODE REVIEW

**Comment by moewew** [comment]
Thank you very much.

The changes except for the "et al." thing look fine to me.

As for "et al." I'm not too sure. Obviously I know nothing about Finish typography and traditions and I'll happily take your research at face value. But I'm worried about the backwards compatibility implications. "et al." is one of these strings that pops up *very* often. Changing it from one version to another will have drastic impact on people's documents. So I think I'd rather not go for that. (Anecdotically, it appears to me that the first change many German-speaking users of `biblatex` make is to change "u.a." to "et al.". So arguably a number of German users and style guides would want `biblatex` to go back to "et al.", which for the same reasons I'd really rather not do.)

**Comment by valtterikantanen** [comment]
Thank you for getting back to me on this.

I completely understand your concerns regarding the backwards compatibility implications of changing "et al." to "ym.". It's a valid point that such a change could significantly impact users' documents, especially given how frequently this term appears.

It's clear that while some Finnish speakers favor "ym.", there remains a significant portion of the community that prefers the traditional "et al.".

With this in mind, I have made a new commit reverting "ym." back to "et al.".

**Comment by moewew** [comment]
Thanks for rolling back the "et al." change.

Changes to translations are always a bit tricky, because in theory any change goes against backwards compatibility. So generally it's safer to stick to fixing obvious and clear mistakes and maybe make some changes to the odd lesser-used string, but "et al." is too ubiquitous  for that. Especially if there are still people who like the status quo.

File: tex/latex/biblatex/lbx/finnish.lbx [hunk 1]
```diff
@@ -432,18 +432,18 @@
   ofseries         = {{sarjassa}{sarjassa}},
   number           = {{numero}{nro}},
   chapter          = {{luku}{luku}},
-  bathesis         = {{tutkielma}{tutkielma}},% FIXME: unsure
-  mathesis         = {{tutkielma}{tutkielma}},% FIXME: unsure
-  phdthesis        = {{tohtorinv\"ait\"oskirja}{tohtorinv\"ait\"oskirja}},
-  candthesis       = {{kanditaatintutkielma}{kanditaatintutkielma}},% Literal translation of "Candidate thesis".
+  bathesis         = {{kandidaatintutkielma}{kandidaatintutkielma}},
+  mathesis         = {{pro gradu -tutkielma}{pro gradu -tutkielma}},
```
Comment (moewew) [inline tex/latex/biblatex/lbx/finnish.lbx:436]: Before we merge: Are the space and hyphen as intended here?

**Pull request merged**